presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -753,6 +754,24 @@ public PlanWithProperties visitValues(ValuesNode node, Context context)
         }
 
         @Override
+        public PlanWithProperties visitExplainAnalyze(ExplainAnalyzeNode node, Context context)
+        {
+            PlanWithProperties child = planChild(node, context.withPreferredProperties(PreferredProperties.any()));
+
+            // if the child is already a gathering exchange, don't add another
+            if ((child.getNode() instanceof ExchangeNode) && ((ExchangeNode) child.getNode()).getType() == ExchangeNode.Type.GATHER) {
+                return rebaseAndDeriveProperties(node, child);
+            }
+
+            // Add an exchange even if this is already coordinator only, because ExplainAnalyze should be in its own stage
```

Comment:
the "even if this is already coordinator only" phrase doesn't make any sense here.  There are only three types of exchanges `GATHER`, `REPARTITION`, and `REPLICATE`.  Coordinator only is a property of `PartitionFunctionBinding`